src/main/java/org/jabref/logic/exporter/CffExporter.java

```diff
@@ -0,0 +1,247 @@
+        Map.entry(StandardEntryType.Dataset, "dataset"),
+        Map.entry(StandardEntryType.Report, "report"),
+        Map.entry(StandardEntryType.Unpublished, "unpublished")
+    );
+
+    public CffExporter() {
+        super("cff", "CFF", StandardFileType.CFF);
+    }
+
+    @Override
+    public void export(BibDatabaseContext databaseContext, Path file, List<BibEntry> entries) throws Exception {
+        Objects.requireNonNull(databaseContext);
+        Objects.requireNonNull(file);
+        Objects.requireNonNull(entries);
+
+        if (entries.isEmpty()) { // Do not export if no entries to export -- avoids exports with only template text
+            return;
+        }
+
+        DumperOptions options = new DumperOptions();
+
+        // Set line width to infinity to avoid line wrapping
+        options.setWidth(Integer.MAX_VALUE);
+
+        // Set collections to be written in block rather than inline
+        options.setDefaultFlowStyle(DumperOptions.FlowStyle.BLOCK);
+        options.setPrettyFlow(true);
+
+        // Set indent for sequences to two spaces
+        options.setIndentWithIndicator(true);
+        options.setIndicatorIndent(2);
+        Yaml yaml = new Yaml(options);
+
+        Map<String, Object> originalData = new LinkedHashMap<>();
+        Map<String, Object> referencesData = new LinkedHashMap<>();
+        Map<String, Object> data = originalData;
+        data.put("cff-version", "1.2.0");
+
+        for (BibEntry entry : entries) {
+            Map<Field, String> entryMap = new HashMap<>(entry.getFieldMap());
+
+            // Mandatory message field
+            String message = entryMap.getOrDefault(StandardField.COMMENT,
+                    "If you use this software, please cite it using the metadata from this file.");
+            data.put("message", message);
+            entryMap.remove(StandardField.COMMENT);
+
+            // Mandatory title field
+            String title = entryMap.getOrDefault(StandardField.TITLE, "No title specified.");
+            data.put("title", title);
+            entryMap.remove(StandardField.TITLE);
+
+            // Mandatory authors field
+            List<Author> authors = AuthorList.parse(entryMap.getOrDefault(StandardField.AUTHOR, ""))
+                                             .getAuthors();
+            List<Map<String, String>> authorsList = parseAuthors(data, authors);
+            entryMap.remove(StandardField.AUTHOR);
+
+            // Type;
+            EntryType entryType = entry.getType();
+            switch (entryType) {
+                case StandardEntryType.Software, StandardEntryType.Dataset ->
+                        data.put("type", entryType.getName());
+                default -> {
+                    if (TYPES_MAP.containsKey(entryType)) {
+                        data.put("references", referencesData);
+                        data = referencesData;
+                        data.put("type", TYPES_MAP.get(entryType));
```

Comment:
Do that directly after the if to be consistent with the case on line 139 (ok, you have to move data two lines up, too)